### ISSUE TITLE
Migrate wave functions to use TimeData

### ISSUE THREAD

**Issue by james-pcdr** [python-can-define-radio/pcdr#22]

**Comment by james-pcdr** [comment]
```
def ook_modulate_at_frequency
def makeComplexWave_numsamps
def makeRealWave_numsamps
def makeComplexWave_time
def makeRealWave_time
def multiply_by_complex_wave
def multiply_by_real_wave
```

Note, don't change `make_fft`

**Comment by james-pcdr** [comment]
Will be fixed by #46
